[Scaffolder Bitbucket Server] Use repository default branch as target branch instead of hardcoded targetBranch (#25031)
backstage/backstage: .changeset/fair-rockets-leave.md

---
'@backstage/plugin-scaffolder-backend-module-bitbucket-server': patch
---

> Instead of using hardcoded targetBranch we are fetching the default branch from Bitbucket repository.

Reviewer: ```suggestion
Instead of using hardcoded `targetBranch` now fetch the default branch from Bitbucket repository.
```

> This prevents from errors when no targetBranch is provided and the default repository branch is different from `master`, for example: `main`.

Reviewer: ```suggestion
This prevents from errors when no `targetBranch` is provided and the default repository branch is different from `master`, for example: `main`.
```
